qiskit/pulse/macros.py

```diff
@@ -104,3 +112,173 @@ def measure_all(backend) -> Schedule:
+    express the entire pulse envelope. This chunking allows the pulse controller to compress
+    the very long flat-top part into a single short constant pulse in the waveform memory,
+    and thus it can reduce waveform memory footprint for a single job.
+
+    Chunked pulse can be parameterized except for the `duration`.
+
+    .. note::
+
+        A chunked pulse may increase the payload size because rising and falling edges
+        are expressed by the raw samples rather than in the parametric form.
+        When required chunk, i.e. number of repeated constant pulse, is less than the
+        `min_chunk_number` limit, this macro function plays a single
+        :class:`.GaussianSquare` pulse instead of generating a chunked pulse schedule.
+
+    .. see_also::
+        :class:`.GaussianSquare` for definition of the pulse envelope.
+
+    Args:
+        duration: Pulse length in terms of the sampling period `dt`.
+            This duration must be an integer numer. Parameterization is not acceptable.
+            Duration may be rounded to match the chunk size and ramp durations
+            with the device timing constraint of the pulse granularity.
+        amp: The magnitude of the amplitude of the Gaussian and square pulse.
+            Complex amp support will be deprecated.
+        angle: The angle of the complex amplitude of the pulse.
+        sigma: A measure of how wide or narrow the Gaussian risefall is.
+        risefall_sigma_ratio: The ratio of each risefall duration to sigma.
+        channel: A pulse channel to play this pulse.
+        granularity: A device timing constraint for the pulse granularity.
+        name: Display name for this pulse envelope.
+        limit_amplitude: If ``True``, then limit the amplitude of the
+            waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+        chunk_size: Size of a single chunk in units of dt. This must satisfy the
+            device timing constraint for the pulse granularity.
+        min_chunk_number: Minimum chunk number to enable the chunk division.
+            This macro plays a single :class:`.GaussianSquare` pulse when
+            the pulse duration is sufficiently short.
+
+    Returns:
+        ScheduleBlock of the chunked GaussianSquare pulse sequence.
+
+    Raises:
+        PulseError: When duration is parameterized.
+    """
+    # TODO this must be called from proper pulse compiler once implemented.
+    #  i.e. Standard GaussianSquare is implicitly replaced with this based on a compiler option.
+    #  Experimentalist doesn't need to take care of chunking like this.
+
+    if not isinstance(duration, int):
+        raise PulseError(
+            "Parameterized duration cannot be applied to the chunked pulse. "
+            "Specify an integer duration in units of dt."
+        )
+
+    if chunk_size % granularity != 0:
+        _new_size = granularity * int(chunk_size / granularity)
```

Comment:
Should we worry about `min_length` here or just hope chunk_size is larger than it?